### PR TITLE
Fix/alessio

### PR DESCRIPTION
**Pull request by AlessioVerardo**
The first PR about bug fixes. Nevertheless, this PR also adds two new functionalities, i.e. the ability to see the status of your event edit requests if you are an activity provider. We also work with @Feez2403 on a clean way to delete event (by canceling MaterialRequests, EventEdits, Events for instance). 

I also fix some bugs as well as tests, here is a small list: 
- The profile informations can now be correctly set and update in the database 
- All current profile are removed from all the database methods as we won't have time to manage database access at the app level
- Disable dark mode which introduces small problems in the toolbar which I didn't have time to fix for now (color black and the arrow is strange)
- Disable landscape mode

And I also refactor the strings folder so that they are more readable.

Do not hesitate to make any feedback.
We will wait for the PR of Neygo to be merged before merging my branch. 

Cheers,

### CODE REVIEW

**Comment by Rygone** [comment]
good idea to remove `userAccess: UserProfile?` from database as we do the check before

File: app/src/main/java/com/github/sdpteam15/polyevents/model/entity/Event.kt [hunk 1]
```diff
@@ -162,19 +162,20 @@ data class Event(
...
-            return ""
+        return if (startTime == null) {
+            ""
         } else {
             //return SimpleDateFormat("k:mm", Locale.getDefault()).format(startTime)
             val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("k:mm")
-            return startTime.format(formatter)
+            startTime.format(formatter)
         }
     }
```
Comment (Rygone) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/entity/Event.kt:172]: you can do better
with 
`import com.github.sdpteam15.polyevents.helper.HelperFunctions.apply`

```
fun formattedStartTime() =
        startTime.apply("") { it.format(DateTimeFormatter.ofPattern("k:mm")) }
```

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/DatabaseHelper.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import com.github.sdpteam15.polyevents.model.entity.MaterialRequest
+import com.github.sdpteam15.polyevents.model.entity.MaterialRequest.Status.*
+import com.github.sdpteam15.polyevents.model.entity.Zone
+import com.github.sdpteam15.polyevents.model.observable.Observable
+import com.github.sdpteam15.polyevents.model.observable.ObservableList
+
+/**
+ * The goal of this objects is to create function to nested relation from the database
+ */
+object DatabaseHelper {
```
Comment (Rygone) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/DatabaseHelper.kt:16]: Good idea add a helper for the database

**Review by Rygone — APPROVED** [review]

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/DatabaseHelper.kt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    /**
+     * Delete an event and all its material request and event edits from the database
+     * @param e the event we want to delete
+     */
+    fun deleteEvent(e: Event) {
```
Comment (Cell00phane) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/DatabaseHelper.kt:22]: So every time we delete an event we should use this method right?

File: app/src/main/res/layout/card_my_event_edit_request.xml [hunk 4]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_weight="1">
+
+    <androidx.cardview.widget.CardView
```
Comment (Cell00phane) [inline app/src/main/res/layout/card_my_event_edit_request.xml:10]: You can refactor out card view and make it the root view

File: app/src/main/res/layout/card_my_event_edit_request.xml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        android:id="@+id/id_event_edit_request_card"
```
Comment (Cell00phane) [inline app/src/main/res/layout/card_my_event_edit_request.xml:11]: You can refactor out cardview and make it the root view

File: app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt [hunk 6]
```diff
@@ -30,10 +30,11 @@ class UserDatabaseTest {
...
     fun updateUserInformation() {
         val user = UserEntity("uid")
-        val userAccess = UserProfile()
 
         HelperTestFunction.nextSetEntity { true }
-        mockUserDatabase.updateUserInformation(user, userAccess)
+        mockUserDatabase.updateUserInformation(user)
+            .observeOnce { assert(it.value) }.then.postValue(false)
+        mockUserDatabase.updateUserInformation(user)
             .observeOnce { assert(it.value) }.then.postValue(false)
```
Comment (Cell00phane) [inline app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt:38]: Why are there two such duplicate instructions?

File: app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt [hunk 7]
```diff
@@ -63,10 +64,11 @@ class UserDatabaseTest {
...
         val isInDb = Observable<Boolean>()
         val uid = "uid"
-        val userAccess = UserProfile()
 
         HelperTestFunction.nextGetEntity { true }
-        mockUserDatabase.inDatabase(isInDb.observeOnce { assert(it.value) }.then, uid, userAccess)
+        mockUserDatabase.inDatabase(isInDb.observeOnce { assert(it.value) }.then, uid)
+            .observeOnce { assert(it.value) }.then.postValue(false)
+        mockUserDatabase.inDatabase(isInDb.observeOnce { assert(it.value) }.then, uid)
             .observeOnce { assert(it.value) }.then.postValue(false)
```
Comment (Cell00phane) [inline app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt:72]: Here as well

**Review by Cell00phane — CHANGES_REQUESTED** [review]
Very nice refactoring and very good new methods for the database. 
However,  just a small thing: I think you could have kept all the strings in one single file strings.xml. Sure it's more readable but I'm not sure if it messes with the Android strings resources when someone would like to translate our app later.

File: app/src/main/res/layout/card_my_event_edit_request.xml (hunk 5 above)
Comment (AlessioVerardo) [reply app/src/main/res/layout/card_my_event_edit_request.xml:11]: Yes, it's done !

File: app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt (hunk 6 above)
Comment (AlessioVerardo) [reply app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt:38]: I think, I made a mistake while merging :)

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/DatabaseHelper.kt (hunk 3 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/helper/DatabaseHelper.kt:22]: Yes, completely

File: app/src/main/java/com/github/sdpteam15/polyevents/model/entity/Event.kt (hunk 1 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/model/entity/Event.kt:172]: I dont think it is pertinent

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/DatabaseHelper.kt (hunk 2 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/helper/DatabaseHelper.kt:16]: Could be a nice idea, but we made a democraty vote on pro and cons, and you lose :)

**Review by Cell00phane — APPROVED** [review]

**Pull request merged**